probe/probe.go

```diff
@@ -225,25 +228,44 @@ ForLoop:
 			t.Controls = report.Controls{}
 		})
 	}
-	if err := p.publisher.Publish(rpt); err != nil {
-		log.Infof("Publish: %v", err)
-	}
+	return rpt
 }
 
 func (p *Probe) publishLoop() {
 	defer p.done.Done()
 	pubTick := time.Tick(p.publishInterval)
+	publishCount := 0
+	var lastFullReport report.Report
 
 	for {
+		var err error
 		select {
 		case <-pubTick:
-			p.drainAndPublish(report.MakeReport(), p.spiedReports)
+			rpt := p.drainAndSanitise(report.MakeReport(), p.spiedReports)
+			fullReport := (publishCount % p.ticksPerFullReport) == 0
+			if !fullReport {
+				rpt.UnsafeUnMerge(lastFullReport)
+			}
+			err = p.publisher.Publish(rpt)
+			if err == nil {
+				if fullReport {
+					lastFullReport = rpt
+				}
+				publishCount++
+			} else {
+				// If we failed to send then drop back to full report next time
+				publishCount = 0
+			}
 
 		case rpt := <-p.shortcutReports:
-			p.drainAndPublish(rpt, p.shortcutReports)
+			rpt = p.drainAndSanitise(rpt, p.shortcutReports)
+			err = p.publisher.Publish(rpt)
 
 		case <-p.quit:
 			return
 		}
+		if err != nil {
+			log.Infof("Publish: %v", err)
```

Comment:
probably, but that's copied from the existing code.

Reply:
On further reflection, failure to publish can be caused by temporary network glitch, or slowness at the backend.  One single event does not justify a warning from Scope; it will not be noticeable by the user.

A series of failures is worth warning about, but that would be a separate issue.